contracts/strategies/BaseStrategy.sol

```diff
@@ -56,63 +67,31 @@ abstract contract BaseStrategy is IStrategy {
     /// ====================================
 
     /// @notice Initializes the allocation strategy
-    /// @param _allo Address of the Allo contract
     /// @param _identityId Id of the identity
     /// @param _poolId Id of the pool
     /// @param _data The data to be decoded
-    /// @dev This function is called internally by the strategy
-    function __BaseStrategy_init(
-        string memory _strategyIdentifier,
-        address _allo,
-        bytes32 _identityId,
-        uint256 _poolId,
-        bytes memory _data
-    ) internal {
-        if (initialized) {
-            revert BaseStrategy_STRATEGY_ALREADY_INITIALIZED();
-        }
-
-        if (_allo == address(0)) {
-            revert BaseStrategy_INVALID_ADDRESS();
-        }
-
-        initialized = true;
+    /// @dev This function is called by Allo.sol
+    function initialize(bytes32 _identityId, uint256 _poolId, bytes memory _data) external virtual onlyAllo {
+        require(_identityId != bytes32(0), "invalid identity id");
+        require(identityId == bytes32(0), "already initialized");
 
-        _setStrategyIdentifier(_strategyIdentifier);
-
-        allo = Allo(_allo);
         identityId = _identityId;
         poolId = _poolId;
-
-        emit Initialized(_allo, _identityId, _poolId, _data);
-    }
-
-    /// @notice Get the identity id
-    /// @return bytes32 The identity id
-    function getIdentityId() external view returns (bytes32) {
-        return identityId;
-    }
-
-    /// @notice Get the pool id
-    /// @return uint256 The pool id
-    function getPoolId() external view returns (uint256) {
-        return poolId;
     }
 
-    /// @notice Get the Allo address
-    /// @return address The Allo address
-    function getAllo() external view returns (address) {
-        return address(allo);
-    }
-
-    /// @notice Returns the strategy identifier
-    function getStrategyIdentifier() external view returns (bytes32) {
-        return STRATEGY_IDENTIFIER;
-    }
-
-    /// @notice sets the strategy identifier
-    /// @param _strategyIdentifier the strategy identifier
-    function _setStrategyIdentifier(string memory _strategyIdentifier) internal {
-        STRATEGY_IDENTIFIER = keccak256(abi.encode(_strategyIdentifier));
+    function skim(address _token) external virtual override onlyAllo {
+        (,, address token, uint256 amount,,,) = allo.pools(poolId);
+        uint256 fundedBalance;
+        if (token == _token) {
+            fundedBalance = amount;
+        }
+        uint256 balance = IERC20(_token).balanceOf(address(this));
+        if (balance > fundedBalance) {
+            uint256 excessFunds = balance - fundedBalance;
+            uint256 bounty = (excessFunds * allo.feeSkirtingBounty()) / allo.FEE_DENOMINATOR();
+            excessFunds -= bounty;
+            IERC20(_token).transfer(treasury, excessFunds);
+            IERC20(_token).transfer(msg.sender, bounty);
```

Comment:
`msg.sender` will always be Allo
we need to pass the senders address

Reply:
And we should introduce a skim event

Reply:
We should also introduce safeTransfer again

Reply:
- Introduced safeTransfer using the Transfer library
- Added Skim event
- Removed `onlyAllo`, this should be called directly, not via Allo